### ISSUE TITLE
[Feature Request] Free look camera implementation

### ISSUE THREAD

**Issue by GeekyGami** [PCSX2/pcsx2#3297]
Not only Dolphin, PPSSPP has freelook camera.
This has been requested many times since 2016 and beforehand;

A freelook camera would be primordial to any form of VR implementation.

Considering that there's already graphics plugins that permit SBS/3D/depth buffers, such as GS3DX, this is the only thing truly missing, beyond a method of hiding unwanted hud/graphical elements and proper VR support.

Games in third person are actually doing quite well in VR, if games like Chronos are any indication.
Playing Wind Waker in VR is an experience like no other.

Kingdom Hearts 1&2 Final Mix + and Shadow of Colossus would be absolutely phenomenal for this.

As for culling, AR codes can handle that.

... Anyways, point being, freelook camera would really make it a lot easier to implement VR to the entire emulator.

(Let's not mention that some channels that thrive from out of bounds discoveries could also use it.)

**Comment by JohnHolmesII** [comment]
No.

**Comment by RedPanda4552** [comment]
The difference is the Gamecube, Wii, and PSP are not the PS2. These consoles have wildly different architectures that allow this to be done with a much lower level of difficulty and with nearly perfect reproduction across all games. 

The PS2 handled all 3D processing (prior to texture mapping/rasterization) on VU0 and VU1 coprocessors which aside from some tech demos provided by Sony had no standardized procedures or functions aside from the ASM level instructions the chips themselves executed. Developers were instead encouraged to write their own microprograms from scratch, and this makes 3D processing almost completely unique to every single game in the PS2 library.

As such, not a soul has come by the PCSX2, Play!, or Dobiestation projects and had the slightest clue of how to reliably reproduce a freecam implementation for the PS2, without it being a game specific hack, for every single PS2 game known to mankind. If there is such a person, they are in hiding and a lot of us would like them to share their knowledge with us. But unfortunately, no such person has come out of hiding yet.

In short, we won't be doing this for a very, very, very long time, if ever. Sorry to disappoint, but it is plain and simply a technological hurdle that no one has a solution to.

**Comment by PSI-Rockin** [comment]
While freecam would be technically possible, it would require per-game patches for the reasons RedPanda has mentioned. Said patches would require a significant amount of time to reverse the game's engine, which means that only someone talented with enough dedication to a single game could do it.

**Comment by bnthomason** [comment]
> Not only Dolphin, PPSSPP has freelook camera.

I don't know where you got this PPSSPP freelook camera, except maybe from GLiIntercept, which doesn't work properly, but as far as I know, there is no PPSSPP freelook camera. If you know of one such build, can you message it to me on discord? Thanks.

Dracoceros #8799

**Comment by Laggamer2005** [comment]
He Has a point. DEW IT!
